ctapipe/io/containers.py

```diff
@@ -78,6 +77,7 @@ class RawCameraContainer(Container):
                             "integrated ADC data (n_pixels)"))
     adc_samples = Item(Map(), ("map of channel to arrays of "
                                "(n_pixels, n_samples)"))
+    num_samples = Item(int, "number of time samples")
```

Comment:
think of Containers as column definitions of a table.  Would you want to repeat num_samples for every row in the table? If it never changes, then it doesn't go there, but in a "header" container like inst, or in the metadata (in which case it becomes header keywords in the file)